pkg/provider/azure_loadbalancer.go

```diff
@@ -2490,15 +2503,19 @@ func (az *Cloud) getExpectedSecurityRules(wantLb bool, ports []v1.ServicePort, s
 					SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
 						Protocol:             *securityProto,
 						SourcePortRange:      to.StringPtr("*"),
-						DestinationPortRange: to.StringPtr(strconv.Itoa(int(port.Port))),
+						DestinationPortRange: to.StringPtr(strconv.Itoa(int(dstPort))),
 						SourceAddressPrefix:  to.StringPtr(sourceAddressPrefixes[j]),
 						Access:               network.SecurityRuleAccessAllow,
 						Direction:            network.SecurityRuleDirectionInbound,
 					},
 				}
 				if len(destinationIPAddresses) == 1 {
 					// continue to use DestinationAddressPrefix to avoid NSG updates for existing rules.
-					nsgRule.DestinationAddressPrefix = to.StringPtr(destinationIPAddresses[0])
+					dstAddr := destinationIPAddresses[0]
+					if disableFloatingIP {
+						dstAddr = "VirtualNetwork"
```

Comment:
is it safe to whitelist all traffic from the Internet to all vnet?

Reply:
Thank you for your patience. When floating IP is disabled for load balancer. IP addresses should map to the backend instances' IPs.

Change has been made to whitelist backend instances' private IPs in nsg rules when floating IP has been disabled for load balancer.